test/sql/copy/s3/url_encode.test

```diff
@@ -93,18 +95,35 @@ statement ok
 SET s3_url_compatibility_mode=true;
 
 statement ok
-COPY test_4 TO 's3://test-bucket-public/url_encode/question?marks?and*stars[and]brackets.parquet' (FORMAT 'parquet');
+COPY test_4 TO '${prefix}test-bucket-public/url_encode/question?marks?and*stars[and]brackets.parquet' (FORMAT 'parquet');
 
 query I
-SELECT * FROM "s3://test-bucket-public/url_encode/question?marks?and*stars[and]brackets.parquet" LIMIT 1;
+SELECT * FROM "${prefix}test-bucket-public/url_encode/question?marks?and*stars[and]brackets.parquet" LIMIT 1;
 ----
 4
 
+statement error
+SELECT * FROM 's3://test-bucket/whatever';
+----
+Catalog Error: Table with name s3://test-bucket/whatever does not exist!
+
+statement error
+SELECT * FROM 'gcs://test-bucket/whatever';
+----
+Catalog Error: Table with name gcs://test-bucket/whatever does not exist!
+
+statement error
+SELECT * FROM 'r2://test-bucket/whatever';
```

Comment:
I wonder if these tests are also executed on Windows, if they are then specifying the full path (including slashes) in the expected error might break when tested on Windows

Reply:
@Tishj I think this should work on windows? the slashes in http urls are `/` regardless of platform?

@chrisiou I have another comment with these tests though. The idea behind this test was to confirm that the generated http urls are correct. So for example when using the settings from `source ./scripts/set_s3_test_server_variables.sh`, when should be able to test this:


```
SELECT * FROM 'gcs://test-bucket/whatever.parquet';
Error: IO Error: Unable to connect to URL "http://test-bucket.duckdb-minio.com:9000/whatever.parquet": 404 (Not Found)
```
Note that the endpoint is now set to `duckdb-minio.com:9000` as defined by the env variable from the script. Now when I run the following:

```
D set s3_endpoint='s3.some.random.endpoint.com';
D SELECT * FROM 'gcs://test-bucket/whatever.parquet';
Error: IO Error: Connection error for HTTP HEAD to 'http://test-bucket.s3.some.random.endpoint.com/whatever.parquet'
```
You can see that the error message can show us we're constructing the correct urls here. Could you rewrite the test in such a way that this test confirm we are using the right endpoint default with each prefix?

Reply:
@samansmink you're right, I'm thinking of local filesystem paths, my bad!